annubes/task.py

```diff
@@ -165,42 +162,32 @@ def generate_trials(
         trials["outputs"] = self._outputs
         return trials
 
-    def _rescale(
+    def _minmaxscaler(
         self,
-        input_: float,
-        coeff: float,
-        min_intensity: float | None = None,
-        max_intensity: float | None = None,
+        input_: NDArray[np.float32],
+        input_range: tuple[float, float] = (0, 1),
     ) -> float:
-        """Rescale `input_` value along (0,`coeff`) if `coeff` is non-zero.
+        """Rescale `input_` array to a given range.
 
         Rescaling happens as follows:
-            coeff * (input_ - min_intensity) / (max_intensity - min_intensity)
+
+            `X_std = (input_ - input_.min()) / (input_.max() - input_.min())`
+            `X_scaled = X_std * (max - min) + min`
+            where min, max = range.
+        The logic is the same as that of `sklearn.preprocessing.MinMaxScaler` estimator. Each array is rescaled to the
+        given range, for each trial contained in `input_`.
+
 
         Args:
-            input_: Value that will be rescaled.
-            coeff: Maximum value after rescaling. If set to 0, `input_` is returned unmodified.
-            min_intensity: Minimum value of the input intensities. Defaults to `min(self.stim_intensities)`
-            max_intensity: Maximum value of the input intensities. Defaults to `max(self.stim_intensities)`
+            input_ (NDArray[np.float32]): Input array of shape (self._ntrials, len(self.time), self.n_inputs).
+            input_range (tuple[float, float]): Desired range of transformed data. Defaults to (0, 1).
```

Comment:
I believe the types don't need to be mentioned in the docstring as long as they are type hinted (see the `def function_with_pep484_type_annotations` example [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html))
```suggestion
            input_: Input array of shape (self._ntrials, len(self.time), self.n_inputs).
            input_range: Desired range of transformed data. Defaults to (0, 1).
```

Reply:
Makes total sense!